tt_metal/detail/tt_metal.hpp

```diff
@@ -112,6 +112,19 @@ namespace tt::tt_metal{
          */
         void CompileProgram(Device *device, Program &program);
 
+
+        /**
+         * Get the base address in L1 of the runtime args for a given processor used by the kernel.
+         *
+         * Return value: uint32_t (base address)
+         *
+         * | Argument     | Description                                                            | Type                          | Valid Range                        | Required |
+         * |--------------|------------------------------------------------------------------------|-------------------------------|------------------------------------|----------|
+         * | kernel       | The kernel that the runtime args are for                               | std::shared_ptr<Kernel>       |                                    | Yes      |
+         */
+
+        uint32_t GetL1ArgBaseAddr(std::shared_ptr<Kernel> kernel);
```

Comment:
Is this an API we want to expose at metal API level? @abhullar-tt

Reply:
Good question. I'm not sure. It was anonymous function only used by WriteRuntimeArgsToDevice() previously. Maybe there is another location for it, if we don't want it to be exposed (did I expose it with my change?)

Reply:
yeah I don't think we need this, we can keep it just in the source

Reply:
Actually, I defined it in the header so it could be used by other parts of code like program.cpp to remove duplication and their incorrect versions (single source of truth):

```
            auto kernel = detail::GetKernel(*this, kernel_id);
            auto dst = detail::GetL1ArgBaseAddr(kernel);
```

(and it could be used by tests too, just realized they also have the same code in `runtime_args.cpp` which doesn't contain whole set, which could be replaced by same function call above.


```
uint32_t get_runtime_arg_addr(std::shared_ptr<Kernel> kernel) {
    uint32_t result_base = 0;
    switch (kernel->processor()) {
        case tt::RISCV::BRISC: {
            result_base = BRISC_L1_ARG_BASE;
           <snip>
```

Better place to puth `GetL1ArgBaseAddr` if not in tt_metal.cpp/.hpp maybe?

Reply:
ah okay that makes sense, I think it should be fine because it is under the `detail` namespace (same space where we have slow dispatch apis that aren't meant to be user facing)